### PR TITLE
Remove cli.factory dependencies from the repo

### PR DESCRIPTION
**Pull request by shweta83**
This PR will remove the dependency of cli.factory from robottelo and cli_factory will be used in place of that.
All the test modules using cli.factory method should be replaced with cli_factory methods as we are deprecating cli.factory.

Old methods -> New methods

make_activation_key -> make_activation_key
make_architecture -> make_architecture
make_content_view -> make_content_view
make_content_view_with_credentials -> content_view_with_credentials
make_content_view_filter -> content_view_filter
make_content_view_filter_rule -> content_view_filter_rule
make_discoveryrule -> discoveryrule
make_content_credential -> make_content_credential
make_location -> make_location
make_model -> model
make_partition_table -> make_partition_table
make_product -> make_product
make_product_with_credentials -> product_with_credentials
make_product_wait -> make_product_wait
make_repository_with_credentials -> repository_with_credentials
make_role -> role
make_filter -> make_filter
make_scap_policy -> scap_policy
make_subnet -> subnet
make_sync_plan -> sync_plan
make_host -> make_host
make_fake_host -> make_fake_host
make_host_collection -> host_collection
make_job_invocation -> job_invocation
make_job_invocation_with_credentials -> Removed as not used anywhere
make_job_template -> job_template
make_user -> user
make_usergroup -> usergroup
make_usergroup_external --> usergroup_external
make_compute_resource -> compute_resource
make_org -> make_org
make_org_with_credentials -> org_with_credentials
make_realm -> realm
make_reporttemplate -> reporttemplate
make_os -> os
make_scapcontent -> scapcontent
make_domain -> make_domain
make_hostgroup -> hostgroup
make_medium -> make_medium
make_environment -> ?
make_lifecycle_environment -> make_lifecycle_environment
make_tailoringfile -> tailoringfile
make_template -> make_template
make_template_input -> template_input
make_virt_who_config -> virt_who_config
activationkey_add_subscription_to_repo -> activationkey_add_subscription_to_repo
setup_org_for_a_custom_repo -> setup_org_for_a_custom_repo
_setup_org_for_a_rh_repo -> _setup_org_for_a_rh_repo
setup_org_for_a_rh_repo -> setup_org_for_a_rh_repo
add_role_permissions -> add_role_permissions
setup_cdn_and_custom_repositories -> setup_cdn_and_custom_repositories
setup_cdn_and_custom_repos_content -> setup_cdn_and_custom_repos_content
make_http_proxy -> Not used anywhere

### CODE REVIEW

**Review by jyejare — CHANGES_REQUESTED** [review]
We are not suppose to use 'cli/factory' module at all. Everything related helepers should be coming from 'cli_factory' object.

**Comment by jyejare** [comment]
@shweta83 Are you ok to resume on this ? 

Just FYI, Automation FG is planning for this kind of cleanup in Q3, this PR would be considered one of them.

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -452,6 +452,27 @@ def make_partition_table(options=None):
     return create_object(PartitionTable, args, options)
 
 
+def make_proxy(self, options=None):
```
Comment (jyejare) [inline robottelo/cli/factory.py:455]: None of the helpers should be going into this module. Infact all the helpers in this module would be moved under `cli_factory` object/class. The plan is to remove the entire `cli/factory.py` in this PR itself!

Comment (jyejare) [reply robottelo/cli/factory.py:455]: Move this under cli_factory.

File: robottelo/host_helpers/cli_factory.py [hunk 2]
```diff
@@ -25,10 +25,12 @@
 from fauxfactory import gen_url
 
 from robottelo import constants
+from robottelo.cli.activationkey import ActivationKey
```
Comment (jyejare) [inline robottelo/host_helpers/cli_factory.py:28]: No imports this should be accessed through satellite object.

`<sat_obj>.cli.ActivationKey`

File: robottelo/host_helpers/repository_mixins.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 """
-All the Repository classes in this module are supposed to use from sat_object.cli_factory object.
+All the Repository classes in this module are supposed to use from robottelo.cli.factory object.
```
Comment (jyejare) [inline robottelo/host_helpers/repository_mixins.py:2]: Nop the previous one is right !

File: tests/foreman/api/test_location.py [hunk 4]
```diff
@@ -26,6 +26,7 @@
 from fauxfactory import gen_string
 from requests.exceptions import HTTPError
 
+from robottelo.cli.factory import make_proxy
```
Comment (jyejare) [inline tests/foreman/api/test_location.py:29]: No imports ! This should have been called by sat_object like:

`<sat_obj>.cli_factory.make_proxy()`

File: tests/foreman/cli/test_activationkey.py [hunk 5]
```diff
@@ -46,6 +44,8 @@
 from robottelo.constants import PRDS
 from robottelo.constants import REPOS
 from robottelo.constants import REPOSET
+from robottelo.host_helpers.cli_factory import CLIFactory
```
Comment (jyejare) [inline tests/foreman/cli/test_activationkey.py:47]: No importing, the CLIFactory class object `cli_factory` is accessible from satellite object and hence it should should have been called as :

`<sat_obj>.cli_factory.<helper_method>`

File: tests/foreman/cli/test_activationkey.py [hunk 6]
```diff
@@ -46,6 +44,8 @@
...
+from robottelo.host_helpers.cli_factory import CLIFactoryError
```
Comment (jyejare) [inline tests/foreman/cli/test_activationkey.py:48]: Remove this helper from here and move under `robottelo/exceptions.py` module and import from there if not already moved.

File: tests/foreman/cli/test_activationkey.py [hunk 7]
```diff
@@ -73,7 +73,7 @@ def test_positive_create_with_name(module_target_sat, module_entitlement_manifes
 
     :parametrized: yes
     """
-    new_ak = module_target_sat.cli_factory.make_activation_key(
+    new_ak = CLIFactory.CLIFactory.make_activation_key(
```
Comment (jyejare) [inline tests/foreman/cli/test_activationkey.py:76]:
```suggestion
    new_ak = module_target_sat.cli_factory.make_activation_key(
```
The right way of using cli_factory methods.

File: tests/foreman/cli/test_activationkey.py [hunk 8]
```diff
@@ -92,7 +92,9 @@ def test_positive_create_with_description(desc, module_org):
 
     :parametrized: yes
     """
-    new_ak = make_activation_key({'organization-id': module_org.id, 'description': desc})
+    new_ak = CLIFactory.CLIFactory.make_activation_key(
```
Comment (jyejare) [inline tests/foreman/cli/test_activationkey.py:95]: Same as above and everywhere applicable in the PR!

File: tests/foreman/cli/test_activationkey.py [hunk 9]
```diff
@@ -1657,7 +1667,7 @@ def test_positive_ak_with_custom_product_on_rhel6(module_org, rhel6_contenthost,
 
     :bz: 2038388
     """
-    entities_ids = target_sat.cli_factory.setup_org_for_a_custom_repo(
+    entities_ids = setup_org_for_a_custom_repo(
```
Comment (jyejare) [inline tests/foreman/cli/test_activationkey.py:1670]: The previous usage was right !

File: tests/foreman/cli/test_capsule.py [hunk 10]
```diff
@@ -37,7 +38,7 @@ def test_positive_import_puppet_classes(session_puppet_enabled_sat, puppet_proxy
     with session_puppet_enabled_sat as puppet_sat:
         port = puppet_sat.available_capsule_port
         with puppet_sat.default_url_on_new_port(9090, port) as url:
-            proxy = puppet_sat.cli_factory.make_proxy({'url': url})
+            proxy = make_proxy({'url': url})
```
Comment (jyejare) [inline tests/foreman/cli/test_capsule.py:41]: Previous usage was right !

File: tests/foreman/ui/test_activationkey.py [hunk 11]
```diff
@@ -1209,7 +1209,7 @@ def test_positive_ak_with_custom_product_on_rhel6(session, rhel6_contenthost, ta
     :CaseLevel: Integration
     """
     org = target_sat.api.Organization().create()
-    entities_ids = target_sat.cli_factory.setup_org_for_a_custom_repo(
+    entities_ids = setup_org_for_a_custom_repo(
```
Comment (jyejare) [inline tests/foreman/ui/test_activationkey.py:1212]: Previous was right !

**Review by jyejare — CHANGES_REQUESTED** [review]
@shweta83  Looks like the changes made here are doing the opposite of what we are supposed to do :)

Please refer comments or else we can sync up !

File: tests/foreman/ui/test_activationkey.py (hunk 11 above)
Comment (shweta83) [reply tests/foreman/ui/test_activationkey.py:1212]: Some wrong commit happened. I will correct all this.

File: tests/foreman/cli/test_activationkey.py (hunk 7 above)
Comment (shweta83) [reply tests/foreman/cli/test_activationkey.py:76]: Incorrect commit happened here. I will correct it today.

File: tests/foreman/cli/test_satellitesync.py [hunk 12]
```diff
@@ -457,7 +457,7 @@ def class_export_entities(module_org):
     }
 
 
-def _create_cv(cv_name, repo, module_org, publish=True):
+def _create_cv(cv_name, repo, module_org, module_target_sat, publish=True):
```
Comment (jyejare) [inline tests/foreman/cli/test_satellitesync.py:460]: Tests using this function is not updated to pass arg to sat.

File: tests/foreman/cli/test_satellitesync.py [hunk 13]
```diff
@@ -486,7 +486,7 @@ def _create_cv(cv_name, repo, module_org, publish=True):
     return content_view, cvv_id
 
 
-def _import_entities(product, repo, cv, mos='no'):
+def _import_entities(product, repo, cv, sat, mos='no'):
```
Comment (jyejare) [inline tests/foreman/cli/test_satellitesync.py:489]: This function is not used anywhere!

**Review by jyejare — CHANGES_REQUESTED** [review]
Looks like all helper methods from `robottelo/cli/factory.py` are no moved to CLIFactory class in `robottelo/host_helpers/cli_factory.py` but are still being accessed via `target_sat.cli_factory` object. 

Please do that change.

File: tests/foreman/cli/test_lifecycleenvironment.py [hunk 14]
```diff
@@ -181,17 +185,17 @@ class LifeCycleEnvironmentPaginationTestCase:
...
         cls.lces_count = 25
-        cls.org = make_org()
+        cls.org = target_sat.cli_factory.make_org()
         env_base_name = gen_string('alpha')
         last_env_name = ENVIRONMENT
         cls.env_names = [last_env_name]
         for env_index in range(cls.lces_count):
             env_name = f'{env_base_name}-{env_index}'
-            make_lifecycle_environment(
+            target_sat.cli_factorymake_lifecycle_environment(
```
Comment (ogajduse) [inline tests/foreman/cli/test_lifecycleenvironment.py:198]:
```suggestion
            target_sat.cli_factory.make_lifecycle_environment(
```

**Review by ogajduse — CHANGES_REQUESTED** [review]
Partial review - 61 / 73 files reviewed. I will review the rest next week.

File: tests/foreman/cli/test_lifecycleenvironment.py [hunk 15]
```diff
@@ -181,17 +185,17 @@ class LifeCycleEnvironmentPaginationTestCase:
     """Test class for LifeCycle Environment pagination tests"""
 
     @classmethod
-    def setUpClass(cls):
+    def setUpClass(cls, target_sat):
```
[the remark this answers is not in the document]
Comment (ogajduse) [reply tests/foreman/cli/test_lifecycleenvironment.py:188]: Is this something that should be taken care of by Shweta? I would prefer creating a separate issue for this and assigning it to someone from the Phoenix team instead. 
I would like to keep this PR focused on the cli.factory removal.

File: tests/foreman/cli/test_container_management.py [hunk 16]
```diff
@@ -16,15 +16,9 @@
...
-    LifecycleEnvironment,
-    Repository,
-    make_content_view,
-    make_lifecycle_environment,
-    make_product_wait,
-    make_repository,
-)
+from robottelo.cli.contentview import ContentView
+from robottelo.cli.lifecycleenvironment import LifecycleEnvironment
+from robottelo.cli.repository import Repository
```
[the remark this answers is not in the document]
Comment (ogajduse) [reply tests/foreman/cli/test_container_management.py:21]: However I strongly agree with your comment, your concern is not related to the subject of this PR. I think it's a good idea to keep this PR focused only on the cli.factory removal. We can address this concern in some other PR.

**Review by ogajduse — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_remoteexecution.py [hunk 17]
```diff
@@ -1287,7 +1292,7 @@ def test_positive_run_job_on_host_registered_to_pull_provider(
         )
         assert_job_invocation_result(make_user_job['id'], rhel_contenthost.hostname)
         # create a file as new user
-        invocation_command = make_job_invocation(
+        invocation_command = module_target_sat.make_job_invocation(
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_remoteexecution.py:1302]: Updated.

File: tests/foreman/cli/test_contentview.py [hunk 18]
```diff
@@ -2096,7 +2191,7 @@ def test_positive_subscribe_chost_by_id_using_rh_content_and_filters(
             }
         )
 
-        cli_factory.make_content_view_filter_rule(
+        module_target_sat.cli_factory.content_view_filter_rule(
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_contentview.py:2282]: cli_factory has content_view_filter_rule.

File: tests/foreman/cli/test_lifecycleenvironment.py (hunk 14 above)
Comment (shweta83) [reply tests/foreman/cli/test_lifecycleenvironment.py]: Updated.

File: tests/foreman/cli/test_contentview.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (ogajduse) [reply tests/foreman/cli/test_contentview.py:2282]: @shweta83 Can you please share a link to a file with this function highlighted? I can not find it.

File: tests/foreman/cli/test_contentview.py [hunk 19]
```diff
@@ -333,11 +340,13 @@ def test_positive_delete_version_by_id(self, module_org, module_product):
         :CaseImportance: High
         """
         # Create new organization, product and repository
-        new_repo = cli_factory.make_repository({'product-id': module_product.id})
+        new_repo = module_target_sat.cli_factory.make_repository(
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_contentview.py:331]: content-type is a mandatory parameter.

File: tests/foreman/cli/test_contentview.py [hunk 20]
```diff
@@ -1448,26 +1514,27 @@ def test_positive_publish_rh_and_custom_content(
...
         )
         # Associate repos with CV
         ContentView.add_repository({'id': new_cv['id'], 'repository-id': module_rhel_content['id']})
         ContentView.add_repository({'id': new_cv['id'], 'repository-id': new_repo['id']})
-        cvf = cli_factory.make_content_view_filter(
+        cvf = module_target_sat.cli_factory.make_content_view_filter(
             {'content-view-id': new_cv['id'], 'inclusion': 'false', 'type': 'rpm'}
         )
-        cli_factory.make_content_view_filter_rule(
+        module_target_sat.cli_factory.content_view_filter_rule(
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_contentview.py:1591]: cli_factory has make_content_view_filter_rule.

File: tests/foreman/cli/test_contentview.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_contentview.py:2282]: https://github.com/SatelliteQE/robottelo/blob/72485feb6f4b046fc651a8f89ed2fd34d93e57d9/robottelo/host_helpers/cli_factory.py#L99

File: tests/foreman/cli/test_lifecycleenvironment.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_lifecycleenvironment.py]: Updated the method.

File: tests/foreman/cli/test_ostreebranch.py [hunk 21]
```diff
@@ -49,15 +43,17 @@ def ostree_user_credentials():
...
     """
-    org = make_org_with_credentials(credentials=ostree_user_credentials)
-    product = make_product_with_credentials({'organization-id': org['id']}, ostree_user_credentials)
+    org = module_target_sat.cli_factory.org_with_credentials(credentials=ostree_user_credentials)
+    product = module_target_sat.cli_factory.product_with_credentials(
+        {'organization-id': org['id']}, ostree_user_credentials
+    )
     # Create new custom ostree repo
-    ostree_repo = make_repository_with_credentials(
+    ostree_repo = module_target_sat.cli_factory.repository_with_credentials(
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_ostreebranch.py:53]: cli_factory has `repository_with_credentials`

File: tests/foreman/cli/test_satellitesync.py (hunk 13 above)
Comment (shweta83) [reply tests/foreman/cli/test_satellitesync.py:500]: Updated

File: tests/foreman/cli/test_contentview.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (ogajduse) [reply tests/foreman/cli/test_contentview.py:2282]: Gotcha, I see it now. Thanks.

File: tests/foreman/cli/test_contentview.py [hunk 22]
```diff
@@ -1205,26 +1258,27 @@ def test_positive_promote_rh_and_custom_content(
...
```
[the remark this answers is not in the document]
Comment (jyejare) [reply tests/foreman/cli/test_contentview.py:1317]: @shweta83 Looks This has not addressed yet !

**Review by jyejare — CHANGES_REQUESTED** [review]
ACK pending very few suggestions pending and super() function usage / testing.

**Comment by jyejare** [comment]
@shweta83 Looks like not all defs have been moved yet to `cli_factory` from `cli/factory.py`.

**Review by ogajduse — APPROVED** [review]
LGTM, pending Jitendra's comments.

**Comment by shweta83** [comment]
> @shweta83 Looks like not all defs have been moved yet to `cli_factory` from `cli/factory.py`.

@jyejare I have updated the description which will replace old to new methods. All the methods are covered in this PR

File: tests/foreman/cli/test_lifecycleenvironment.py [hunk 23]
```diff
@@ -180,27 +189,27 @@ def test_positve_list_paths(module_org):
...
 
-    @classmethod
-    def setUpClass(cls):
+    @pytest.fixture(scope='class', autouse=True)
+    def class_setup(self, target_sat):
         """Create organization and lifecycle environments to reuse in tests"""
         super().setUpClass()
-        cls.lces_count = 25
-        cls.org = make_org()
+        self.lces_count = 25
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_lifecycleenvironment.py:196]: These are used outside.

File: tests/foreman/cli/test_lifecycleenvironment.py [hunk 24]
```diff
@@ -180,27 +189,27 @@ def test_positve_list_paths(module_org):
 class LifeCycleEnvironmentPaginationTestCase:
     """Test class for LifeCycle Environment pagination tests"""
 
-    @classmethod
-    def setUpClass(cls):
+    @pytest.fixture(scope='class', autouse=True)
+    def class_setup(self, target_sat):
         """Create organization and lifecycle environments to reuse in tests"""
         super().setUpClass()
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/cli/test_lifecycleenvironment.py:195]: Yes. The test cases were passed.

**Comment by shweta83** [comment]
> @shweta83 Could you then please remove`cli/factory.py` module from here? We can merge it post that !

Removed.

**Review by jyejare — APPROVED** [review]

**Pull request merged**